DOC: unambiguous np.histogram dtype description

hmeine opened the pull request:
As a follow-up of #25403, let's be more specific than "dictates the
dtype" because that could still mean that there is a mapping applied
such as weights uint8 -> acc uint64, weights float -> acc double, ...

That would make so much sense that people could expect that and thus
read it into the documentation, I am afraid.

Skipping all CI because this is a documentation correction:
[skip ci]

<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

hmeine committed:
DOC: unambiguous np.histogram dtype description

As a follow-up of #25403, let's be more specific than "dictates the
dtype" because that could still mean that there is a mapping applied
such as weights uint8 -> acc uint64, weights float -> acc double, ...

That would make so much sense that people could expect that and thus
read it into the documentation, I am afraid.

ngoldbaum commented on numpy/lib/_histograms_impl.py:
```diff
@@ -725,7 +726,9 @@ def histogram(a, bins=10, range=None, density=None, weights=None):
         The values of the histogram. See `density` and `weights` for a
-        description of the possible semantics.
+        description of the possible semantics.  If `weights` are given,
+        ``hist.dtype`` will be taken from `weights`, otherwise it will
+        hold large integers.
```
```suggestion
        ``hist.dtype`` will be taken from `weights`.
```

Either delete this or rephrase a bit, "hold large integers" is ambiguous. I also think the explanation you've added in the `weights` description is sufficient, no need to repeat with different wording it here.

hmeine replied:
I looked at it again and just followed your suggestion / pushed an updated branch. (For a clean history, that second commit could be a "fixup" commit squashed into the first, but I don't know how the numpy merge process looks like.)

hmeine committed:
DOC: remove unspecific and superfluous new part again

following ngoldbaum's suggestion, "large integers" was not a good description
and not necessary here.

rossbar reviewed (approved):
Thanks @hmeine for the updates!
